src/shared/services/challenges.js

```diff
@@ -252,7 +253,10 @@ export function normalizeMarathonMatch(challenge, username) {
  * @private
  */
 async function checkError(res) {
-  if (!res.ok) throw new Error(res.statusText);
+  if (!res.ok) {
```

Comment:
Hmm... I think we should show API error only in case of 5xx error statuses. 4xx is either user doing something wrong (trying access something non-existing, or he is not authorized to access), either a bug in the frontend code. These we don't really want to show with icon, but with error modal messages (I mean, my reasoning about it is that server problems are kind of a permanent condition of the server, until somebody restarts it or somehow fixes the bug; while 4xx seems to be one-time-things, so if you just don't do the same bad thing again, you won't meet the error again).